### PR TITLE
chore(#158): added story of Notification to storybook

### LINKED ISSUE #158: Form improvements
### Description

A clear and concise description of what the issue is about.

### Tasks

Include specific tasks in the order they need to be done. Include links to specific lines of code where the job should happen.

- [x] Replace alert in form with proper error handling.
       5xx errors (server issues): Display these using the error component.
       4xx errors (client-side issues):
          400 (validation errors): Display errors next to the specific field that caused the issue.
         Example: If the initials field is incorrect, show the error directly below it.
      Other 4xx errors: Show these at the bottom of the form, just above the "Sign Up" button.

- [x] Disable the submit and retry button when fetching is in progress.
       Prevent multiple form submissions by disabling the button while the request is ongoing.

- [x] Solve error in browser terminal related to the auth form.
       Problem: Fix the error in the browser's developer console related to the authentication form.

- [x] Generate a unique mutation ID for each request.
       Ensure every GraphQL mutation has a clientMutationId for tracking purposes. 

- [x] Create stories for Notification component.
      
### Acceptance Criteria

Include specific criteria for how to prove that this issue will be done

- [x] The submit and retry button are disabled while the fetch request is in progress.
- [x] Each mutation request includes a unique clientMutationId.
- [x]  The new code is covered by unit and integration tests.

Remember to use helpful labels and milestones

### PR DESCRIPTION
**Pull request by Vladomira**
## Description

This PR addresses multiple improvements and fixes for the authentication form and its related components. It introduces robust error handling, prevents multiple submissions, and ensures mutation tracking through unique IDs. Additionally, it improves the Notification component by introducing Storybook stories for better documentation and testing.

## Related Issue
[issue](https://github.com/VilnaCRM-Org/website/issues/158)

## Motivation and Context

This change improves the reliability and user experience of the authentication form. It replaces the use of alert() with contextual error messages, improves form behavior during submissions by disabling buttons to prevent duplicates, and addresses a console error related to the auth form. Additionally, it ensures all GraphQL mutations are trackable via clientMutationId, and improves component documentation by adding stories for the Notification component.

## How Has This Been Tested?
- Manual testing in the browser:
  - Verified that field-specific errors (400) show inline.
  - Verified that server errors (5xx) are displayed in the general error component.
  - Ensured other 4xx errors appear near the submit button.
  - Confirmed that the submit and retry buttons are disabled during requests.
  - Checked that the error in the browser console is no longer present.

- Unit and integration tests:
  - Added tests to validate error rendering logic and button state.

- Confirmed each mutation includes a clientMutationId.
  - Storybook:

Verified new stories for the Notification component render correctly with various props.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING.md**](https://github.com/VilnaCRM-Org/frontend-ssr-template/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] You have only one commit (if not, squash them into one commit).

## Summary by CodeRabbit

- **New Features**
  - Added interactive Storybook stories for the Notification component.
  - Enhanced notification and error handling with more descriptive, categorized, and localized error messages for sign-up and network issues.
  - Improved password and full name validation with clearer, more user-friendly feedback.
  - Added comprehensive favicon and web manifest support for better browser and device integration.
  - Dynamic client ID generation for signup submissions to improve uniqueness.

- **Bug Fixes**
  - Fixed notification fallback to show error notifications correctly for unknown types.
  - Disabled retry buttons during loading states to prevent duplicate actions.
  - Improved form validation logic to provide accurate and specific error messages.

- **Style**
  - Refined notification component styles for better visual hierarchy, layering, and responsiveness.

- **Chores**
  - Added `uuid` dependency for unique ID generation.
  - Improved Docker and Makefile configurations for optimized builds and memory leak testing.
  - Streamlined Apollo error handling to return error messages without side effects.

- **Documentation**
  - Updated localization files for English and Ukrainian with clearer validation and error messages.
  - Removed outdated package management documentation from README.

- **Tests**
  - Refactored and expanded test coverage for form validation, notification, and error handling.
  - Improved test utilities and fixtures for more reliable and maintainable testing.
  - Enhanced mocking and standardized rendering helpers in tests.

### CODE REVIEW

File: src/features/landing/components/Notification/NotificationFallback.tsx [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
+        type="button"
+        size="medium"
+        fullWidth
+        onClick={onHandleClose}
+      >
+        <Typography component="span" sx={styles.messageButtonText}>
+          {t('notifications.success.button')}
+        </Typography>
+      </UiButton>
+    </>
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/features/landing/components/Notification/NotificationFallback.tsx]: 'notifications.success.button' - has text 'Go back' thats why i use it. maybe i should rename it in the object

File: src/features/landing/components/Notification/Notification.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  args: {
+    type: NotificationStatus.SUCCESS,
+    isOpen: true,
+  },
+  render: args => (
+    <div style={storyContainerStyle}>
+      <Notification {...args} />
+    </div>
+  ),
+};
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/features/landing/components/Notification/Notification.stories.tsx]: actually my default component has  onRetry: { action: 'onRetry' }, so i can see that as Action in the storybook when i click retry button

**Comment by Vladomira** [comment]
@coderabbitai

File: MemoryLeak.Dockerfile [hunk 3]
```diff
@@ -1,23 +1,19 @@
...
     && npm install -g pnpm@10.6.5
 
-WORKDIR /app
-COPY package.json pnpm-lock.yaml checkNodeVersion.js ./
+ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser
 
+WORKDIR /app
 
+COPY package.json pnpm-lock.yaml ./
 RUN pnpm install
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply MemoryLeak.Dockerfile]: we don't add comments. code should be descriptive by itself

File: pages/_document.tsx [hunk 4]
```diff
@@ -5,7 +5,27 @@
...
+        <link rel="icon" type="image/svg+xml" sizes="512x512" href="/layout/favicon/512х512.svg" />
+        <link
+          rel="icon"
+          type="image/svg+xml"
+          sizes="1024x1024"
+          href="/layout/favicon/1024х1024.svg"
+        />
+
+        <link rel="shortcut icon" href="/layout/favicon/32х32.svg" />
+      </Head>
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/_document.tsx]: no, I won't add comments, they separated, that is enough

File: MemoryLeak.Dockerfile [hunk 5]
```diff
@@ -1,23 +1,16 @@
...
 
-RUN rm -rf /tmp/chromium/Singleton* /tmp/chromium/Lock /tmp/chromium/Default/Singleton* && \
-    mkdir -p /tmp/chromium && chown -R root:root /tmp/chromium
-
-RUN mkdir -p /root/.config/chromium/docker-chromium-profile && chown -R root:root /root/.config/chromium
-
-CMD ["sh", "-c", "pkill -o chromium || true && node src/test/memory-leak/runMemlabTests.js"]
+CMD ["tail", "-f", "/dev/null"]
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply MemoryLeak.Dockerfile]: What if add the same health check as was in previous docker files? will it make sense or for current case we need another approach?

[the remark this answers is not in the document]
Comment (Vladomira) [reply MemoryLeak.Dockerfile]: we have common-healthchecks.yml for health check 
and healthcheck in docker-compose.test.yml that is applied for different services

**Comment by Vladomira** [comment]
@coderabbit
please check tabs in Makefile. it seems that help command doesn't work and tabs in the terminal neither.

File: pages/_document.tsx [hunk 6]
```diff
@@ -5,7 +5,31 @@ const mainLanguage: string = process.env.NEXT_PUBLIC_MAIN_LANGUAGE as string;
...
+        <meta name="application-name" content="VilnaCRM" />
+        <meta name="theme-color" content="#ffffff" />
+
+        <link rel="icon" type="image/svg+xml" sizes="16x16" href="/layout/favicon/16x16.svg" />
+        <link rel="icon" type="image/svg+xml" sizes="32x32" href="/layout/favicon/32x32.svg" />
+        <link rel="icon" type="image/svg+xml" sizes="64x64" href="/layout/favicon/64x64.svg" />
+        <link rel="icon" type="image/svg+xml" sizes="96x96" href="/layout/favicon/96x96.svg" />
+        <link rel="icon" type="image/svg+xml" sizes="128x128" href="/layout/favicon/128x128.svg" />
+        <link rel="icon" type="image/svg+xml" sizes="256x256" href="/layout/favicon/256x256.svg" />
+        <link rel="icon" type="image/svg+XML" sizes="512x512" href="/layout/favicon/512x512.svg" />
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/_document.tsx]: please check tabs in Makefile. it seems that help command doesn't work and tabs in the terminal neither.

File: public/layout/favicon/site.webmanifest [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "VilnaCRM",
+  "short_name": "VilnaCRM",
+  "start_url": "/",
+  "display": "standalone",
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply public/layout/favicon/site.webmanifest]: is there anything I should add

File: pages/_document.tsx [hunk 8]
```diff
@@ -1,11 +1,42 @@
 import { Html, Head, Main, NextScript } from 'next/document';
 
-const mainLanguage: string = process.env.NEXT_PUBLIC_MAIN_LANGUAGE as string;
+const mainLanguage: string = process.env.NEXT_PUBLIC_MAIN_LANGUAGE ?? 'en';
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/_document.tsx:3]: my linter doesn't allow me to do that; we need to add a type annotation according to rules in the project.

File: pages/_document.tsx [hunk 9]
```diff
@@ -1,11 +1,42 @@
...
+        <link rel="icon" type="image/svg+xml" sizes="128x128" href="/layout/favicon/128x128.svg" />
+        <link rel="icon" type="image/svg+xml" sizes="256x256" href="/layout/favicon/256x256.svg" />
+        <link rel="icon" type="image/svg+xml" sizes="512x512" href="/layout/favicon/512x512.svg" />
+        <link
+          rel="icon"
+          type="image/svg+xml"
+          sizes="1024x1024"
+          href="/layout/favicon/1024x1024.svg"
+        />
+
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/_document.tsx:29]: i don't have that file in the folder, do i need to add it?

File: pages/_document.tsx [hunk 10]
```diff
@@ -1,11 +1,42 @@
...
 export default function Document(): React.ReactElement {
   return (
     <Html lang={mainLanguage}>
-      <Head />
+      <Head>
+        <meta charSet="utf-8" />
+        <meta name="apple-mobile-web-app-title" content="VilnaCRM" />
+        <meta name="application-name" content="VilnaCRM" />
+        <meta name="theme-color" content="#ffffff" />
+
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/_document.tsx]: could you help me to improve my tests, I need to cover mutation score:
src/features/landing/components/AuthSection/AuthForm/AuthLayout.tsx:25:52
  const [errorText, setErrorText] = React.useState('');
   const [errorText, setErrorText] = React.useState("Stryker was here!");

i tried to add test to AuthLaout (was deleted) and NotificationError but it didn't cover that mutation

File: public/layout/favicon/site.webmanifest [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        },
+        {
+          "src": "/layout/favicon/web-app-manifest-512X512.png",
+          "sizes": "512x512",
+          "purpose": "maskable"
+        }
+      ]
+    }
+  ]
+}
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply public/layout/favicon/site.webmanifest]: show the full code please

File: src/test/testing-library/NotificationError.test.tsx [hunk 12]
```diff
@@ -40,31 +44,53 @@ describe('NotificationError Component', () => {
...
+        onRetry={mockOnRetry}
+        loading={false}
+        errorText="Network error"
+      />
+    );
+
+    expect(screen.getByText('Network error')).toBeInTheDocument();
+  });
+  it('should disable the retry button when loading is true', () => {
+    render(<NotificationError setIsOpen={mockSetIsOpen} onRetry={mockOnRetry} loading />);
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/NotificationError.test.tsx:80]: i get the error if i change that: ESLint: Value must be omitted for boolean attribute `loading` (react/jsx-boolean-value)

**Review by Kravalg — CHANGES_REQUESTED** [review]
Please fix all CI checks before I will be able to review this PR

File: public/layout/favicon/site.webmanifest [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+          "src": "/layout/favicon/web-app-manifest-192x192.png",
+          "sizes": "192x192",
+          "purpose": "any maskable"
+        },
+        {
+          "src": "/layout/favicon/web-app-manifest-512x512.png",
+          "sizes": "512x512",
+          "purpose": "maskable"
+        }
+      ]
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply public/layout/favicon/site.webmanifest]: check please, didn't I add that?

File: docker/apollo-server/server.mts [hunk 14]
```diff
@@ -51,7 +51,7 @@ export const resolvers = {
       validateCreateUserInput(input);
       try {
         const newUser: User = {
-          id: '1',
+          id: input.clientMutationId,
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply docker/apollo-server/server.mts]: please check it again, I've changed it like this:
 id: input.clientMutationId || uuidv4(),

File: pages/i18n/localization.json [hunk 15]
```diff
@@ -311,24 +321,24 @@
...
           },
           "email_input": {
             "label": "E-mail",
             "placeholder": "vilnaCRM@gmail.com",
             "required": "Це поле обов'язкове",
-            "step_error_message": "Email має містити символи '@' та '.'",
-            "invalid_message": "Неправильна електронна адреса"
+            "step_error_message": "Будь ласка, переконайтеся, що email містить '@' та '.'",
+            "invalid_message": "Будь ласка, введіть коректну електронну адресу"
           },
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/i18n/localization.json]: i have in the localization file step_error_message 2 times and no any email_format_error

File: public/layout/favicon/browserconfig.xml [hunk 16]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<browserconfig>
+    <msapplication>
+        <tile>
+            <square150x150logo src="/layout/favicon/microsoft-150x150.png"/>
+            <TileColor>#ffffff</TileColor>
+        </tile>
+    </msapplication>
+</browserconfig>
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply public/layout/favicon/browserconfig.xml]: will this be correct?
```
 <tile>
            <square70x70logo src="/layout/favicon/microsoft-70x70.png"/>
            <square150x150logo src="/layout/favicon/microsoft-150x150.png"/>
            <wide310x150logo src="/layout/favicon/microsoft-310x150.png"/>
            <square310x310logo src="/layout/favicon/microsoft-310x310.png"/>
            <TileColor>#ffffff</TileColor>
        </tile>
```

File: pages/_document.tsx [hunk 17]
```diff
@@ -1,11 +1,44 @@
...
+        <link
+          rel="icon"
+          type="image/svg+xml"
+          sizes="1024x1024"
+          href="/layout/favicon/1024x1024.svg"
+        />
+
+        <link rel="shortcut icon" type="image/svg+xml" href="/layout/favicon/32x32.svg" />
+        <link rel="icon" type="image/png" sizes="32x32" href="/layout/favicon/32x32.png" />
+
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/_document.tsx]: they have different format. please say which one should i save

File: MemoryLeak.Dockerfile [hunk 18]
```diff
@@ -1,23 +1,24 @@
...
 
-WORKDIR /app
-COPY package.json pnpm-lock.yaml checkNodeVersion.js ./
+ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
 
+WORKDIR /app
 
+COPY package.json pnpm-lock.yaml ./
 RUN pnpm install
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply MemoryLeak.Dockerfile]: I won't add --frozen-lockfile as in other Dockerfiles there are no such flag

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,45 +1,29 @@
...
-
-
+- **#28:** redesign "For Whom" block ([#171](https://github.com/VilnaCRM-Org/website/issues/171)) ([cd224fe](https://github.com/VilnaCRM-Org/website/commit/cd224fed3a45a086dc8db1ec89cb6be4d62640fb)), closes [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28) [#28](https://github.com/VilnaCRM-Org/website/issues/28)
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply CHANGELOG.md:17]: I can't change this file. this is automatically generated

File: src/features/landing/helpers/handleApolloError.ts [hunk 20]
```diff
@@ -1,16 +1,25 @@
...
+  denied: t('failure_responses.authentication_errors.access_denied'),
+
+  went_wrong: t('failure_responses.client_errors.something_went_wrong'),
+  unexpected: t('failure_responses.client_errors.unexpected_error'),
 
-type HandleNetworkErrorProps = HandleErrorProps & { networkError: ApolloError['networkError'] };
-type HandleNetworkErrorType = (props: HandleNetworkErrorProps) => void;
+  network: t('failure_responses.network_errors.network_error'),
+  server_error: t('failure_responses.server_errors.server_error'),
+};
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/features/landing/helpers/handleApolloError.ts]: I thought about that and created shared folder in test folder - to separate tests as module and landing. so do you think I should remove that folder to src and reuse?

File: src/test/unit/handleApolloError.test.ts [hunk 21]
```diff
@@ -1,248 +1,121 @@
...
-        setIsNotificationOpen: setIsNotificationOpenMock,
-      });
+    it('should return unexpected error for non-ApolloError', () => {
+      const error: HandleApolloErrorProps = { error: null };
+      expect(handleApolloError(error)).toBe(messages.unexpected);
 
-      expect(setServerErrorMessageMock).toHaveBeenCalledWith('Error 1, Error 2');
+      const notApolloError: HandleApolloErrorProps = { error: {} as ApolloError };
+      expect(handleApolloError(notApolloError)).toBe(messages.unexpected);
     });
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/unit/handleApolloError.test.ts]: i don't have that status (FORBIDDEN) in the handle functions

File: pages/_document.tsx [hunk 22]
```diff
@@ -1,11 +1,44 @@
...
   return (
     <Html lang={mainLanguage}>
-      <Head />
+      <Head>
+        <meta charSet="utf-8" />
+        <meta name="description" content="VilnaCRM platform for customer relationship management" />
+        <meta name="apple-mobile-web-app-title" content="VilnaCRM" />
+        <meta name="application-name" content="VilnaCRM" />
+        <meta name="theme-color" content="#ffffff" />
+        <meta name="msapplication-config" content="/layout/favicon/browserconfig.xml" />
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply pages/_document.tsx:14]: if i add <meta name="viewport" content="width=device-width, initial-scale=1" /> I get warning, so probably I have that in somewhere else

File: src/test/unit/handleApolloError.test.ts (hunk 21 above)
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/unit/handleApolloError.test.ts]: should I add condition for such case? I added your tests and I got  Forbidden Access as a final result. Is it acceptable?

File: src/test/testing-library/AuthLayout.test.tsx [hunk 23]
```diff
@@ -470,16 +364,100 @@ describe('AuthLayout', () => {
...
+    };
+
+    const { getByText } = renderAuthLayout([specialCharsMock]);
+
+    fillForm(longName, specialCharsEmail, complexPassword, true);
+
+    await waitFor(() => {
+      expect(getByText(successTitleText)).toBeInTheDocument();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/AuthLayout.test.tsx]: if i got you right, i should update test 'handles maximum length and special character inputs' to use different edge cases?

File: src/test/testing-library/AuthLayout.test.tsx [hunk 24]
```diff
@@ -123,17 +123,28 @@ const validationInputFields: ValidationFormElement[] = [
...
+
+function renderAuthLayout(mocks: MockedResponse[]): RenderResult {
+  return render(
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <AuthLayout />
+    </MockedProvider>
+  );
+}
+
+const NETWORK_FAILURE: Error = new Error('Failed to fetch');
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/AuthLayout.test.tsx]: what else can i replace to fixtures.ts

[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/AuthLayout.test.tsx]: for now I have utils and constants, should I do something with them, like to make refactor?

[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/AuthLayout.test.tsx]: should i remove constants to fixtures folder?

File: src/test/testing-library/fixtures/errors.ts [hunk 25]
```diff
@@ -0,0 +1,11 @@
...
+export const NETWORK_FAILURE: Error = new Error(networkMessage);
+
+export class ServerError extends Error {
+  public statusCode: number = 500;
+
+  constructor(message = 'Network error') {
+    super(message);
+    this.name = 'ServerError';
+  }
+}
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/fixtures/errors.ts:11]: you wrote it me several times, my linter want me to add type annotation explicitly

File: src/shared/clientErrorMessages.ts [hunk 26]
```diff
@@ -0,0 +1,23 @@
...
+type clientErrorKeys =
+  | 'network'
+  | 'unauthorized'
+  | 'denied'
+  | 'unexpected'
+  | 'went_wrong'
+  | 'server_error';
+
+type GetClientsErrorMessages = () => Record<clientErrorKeys, string>;
+export type ClientErrorMessages = Record<clientErrorKeys, string>;
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/shared/clientErrorMessages.ts]: please check it again, for now I use a function getClientErrorMessages not object errorMessages

[the remark this answers is not in the document]
Comment (Vladomira) [reply src/shared/clientErrorMessages.ts]: are you sure that this is better approach for our case? with use those codes in handle functions and tests, so will it be as convenient as it is for now for developer?

[the remark this answers is not in the document]
Comment (Vladomira) [reply src/shared/clientErrorMessages.ts]: what if we use 
 const CLIENT_ERROR_KEYS :ClientErrorKeys= {
  NETWORK: 'network',
  UNAUTHORIZED: 'unauthorized',
  DENIED: 'denied',
  UNEXPECTED: 'unexpected',
  WENT_WRONG: 'went_wrong',
  SERVER_ERROR: 'server_error',
} as const;

File: src/features/landing/helpers/handleApolloError.ts [hunk 27]
```diff
@@ -25,79 +17,76 @@ export function isServerError(error: unknown): error is { statusCode?: number; m
...
-  if (networkError.message?.includes('Failed to fetch')) {
-    setServerErrorMessage('Network error. Please check your internet connection.');
-    return;
+  if (
+    loweredMessage.includes('failed to fetch') ||
+    loweredMessage.includes('network request failed') ||
+    loweredMessage.includes('fetch failed')
+  ) {
+    return messages.network;
   }
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/features/landing/helpers/handleApolloError.ts]: it looks very bad, what if i change it to
 if (NETWORK_ERROR_PATTERNS.some(pattern => loweredMessage.includes(pattern))) {
    return messages[CLIENT_ERROR_KEYS.NETWORK];
  }

File: MemoryLeak.Dockerfile [hunk 28]
```diff
@@ -1,23 +1,27 @@
...
+    freetype \
+    harfbuzz \
+    ca-certificates \
     ttf-freefont \
-    chromium \
-    python3\
-    make \
-    g++ \
-    && npm install -g pnpm@10.6.5
+    && npm install -g pnpm@10.11.0 \
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply MemoryLeak.Dockerfile]: can you suggest suitable versions?

File: src/features/landing/helpers/handleApolloError.ts [hunk 29]
```diff
@@ -25,79 +23,84 @@ export function isServerError(error: unknown): error is { statusCode?: number; m
...
+      graphQLErrors
+        .map(e => e.message || '')
+        .filter(Boolean)
+        .join(', ') || messages[CLIENT_ERROR_KEYS.UNEXPECTED]
+    );
   }
-  setServerErrorMessage('An unexpected error occurred. Please try again.');
+
+  return messages[CLIENT_ERROR_KEYS.UNEXPECTED];
 };
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/features/landing/helpers/handleApolloError.ts:107]: switch (true)  - why you just use true condition? is it safe?

File: docker-compose.memory-leak.yml [hunk 30]
```diff
@@ -5,5 +5,21 @@ services:
...
+      - MEMLAB_SKIP_WARMUP=true
     init: true
-    shm_size: '256m'
+    shm_size: '1gb'
+    healthcheck:
+      interval: 10s
+      timeout: 5s
+      retries: 3
+      start_period: 30s
+      test: ["CMD", "chromium-browser", "--version"]
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply docker-compose.memory-leak.yml:22]: common-healthchecks.yml this file run when I start prod, but in the file docker-compose.test.yml I don't have memory-leak, so I can't include memory leak to that file.

File: docker-compose.memory-leak.yml [hunk 31]
```diff
@@ -5,5 +5,21 @@ services:
       dockerfile: MemoryLeak.Dockerfile
     volumes:
       - .:/app
+      - memleak_node_modules:/app/node_modules
+    environment:
+      - DISPLAY=:99
+      - PUPPETEER_DISABLE_HEADLESS_WARNING=1
+      - NEXT_PUBLIC_PROD_CONTAINER_API_URL=http://prod:3001
+      - MEMLAB_DEBUG=true
+      - MEMLAB_SKIP_WARMUP=true
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply docker-compose.memory-leak.yml:12]: I thought that I need add env variables directly to docker-compose

File: src/test/testing-library/fixtures/auth-test-helpers.tsx [hunk 32]
```diff
@@ -0,0 +1,134 @@
...
+    !!input.password &&
+    typeof input.password === 'string' &&
+    input.password.length >= 8 &&
+    !!input.clientMutationId &&
+    typeof input.clientMutationId === 'string'
+  );
+};
+export interface ExtendedMockedResponse extends MockedResponse {
+  variableMatcher: (variables: { input: CreateUserInput }) => boolean;
+}
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/fixtures/auth-test-helpers.tsx]: I have tests related to variableMatcher, so now should I rewrite everything?

File: src/test/testing-library/fixtures/auth-test-helpers.tsx [hunk 33]
```diff
@@ -0,0 +1,134 @@
...
+
+type CreateTestInputFn = (overrides?: Partial<CreateUserInput>) => CreateUserInput;
+
+const createTestInput: CreateTestInputFn = (overrides = {}) => ({
+  email: testEmail.toLowerCase(),
+  initials: testInitials,
+  password: testPassword,
+  clientMutationId: '132',
+  ...overrides,
+});
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/fixtures/auth-test-helpers.tsx:72]: in AuthLayout I mock id
jest.mock('uuid', () => ({
  v4: jest.fn(() => '132'),
}));

If I change it to any string it will break my tests

File: src/test/testing-library/fixtures/auth-test-helpers.tsx (hunk 32 above)
[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/fixtures/auth-test-helpers.tsx]: why do I need this If I already have variableMatcher: validateCreateUserInput,
and in some cases I need empty inputs.

[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/fixtures/auth-test-helpers.tsx]: that was made to tests inputs.
 if we wanna make it cleaner and don't extend fulfilledMockResponse, will it be better approach to add new tests that check those inputs and modify those that we already have?

[the remark this answers is not in the document]
Comment (Vladomira) [reply src/test/testing-library/fixtures/auth-test-helpers.tsx]: you know, you was wrong, because the type that I use for fulfilledMockResponse is MockedResponse and it indeed has a variableMatcher
export interface MockedResponse<out TData = Record<string, any>, out TVariables = Record<string, any>> {
    request: GraphQLRequest<TVariables>;
    maxUsageCount?: number;
    result?: FetchResult<Unmasked<TData>> | ResultFunction<FetchResult<Unmasked<TData>>, TVariables>;
    error?: Error;
    delay?: number;
    variableMatcher?: VariableMatcher<TVariables>;
    newData?: ResultFunction<FetchResult<Unmasked<TData>>, TVariables>;
}

**Review by Kravalg — CHANGES_REQUESTED** [review]
Please resolve the conflicts

File: MemoryLeak.Dockerfile [hunk 34]
```diff
@@ -1,23 +1,30 @@
...
+COPY package.json pnpm-lock.yaml checkNodeVersion.js ./
 RUN pnpm install
 
-RUN rm -rf /tmp/chromium/Singleton* /tmp/chromium/Lock /tmp/chromium/Default/Singleton* && \
-    mkdir -p /tmp/chromium && chown -R root:root /tmp/chromium
-
-RUN mkdir -p /root/.config/chromium/docker-chromium-profile && chown -R root:root /root/.config/chromium
-
-CMD ["sh", "-c", "pkill -o chromium || true && node src/test/memory-leak/runMemlabTests.js"]
+CMD ["sh", "-c", "rm -f /tmp/.X99-lock && Xvfb :99 -screen 0 1024x768x16 & sleep infinity"]
```
Comment (Kravalg) [inline MemoryLeak.Dockerfile]: This solution is highly questionable and feels like a workaround

File: docker-compose.memory-leak.yml [hunk 35]
```diff
@@ -5,5 +5,21 @@ services:
       dockerfile: MemoryLeak.Dockerfile
     volumes:
       - .:/app
+      - memleak_node_modules:/app/node_modules
```
Comment (Kravalg) [inline docker-compose.memory-leak.yml]: Why do we need this line ?

**Review by Kravalg — CHANGES_REQUESTED** [review]

File: MemoryLeak.Dockerfile [hunk 36]
```diff
@@ -2,22 +1,43 @@
...
+    harfbuzz=9.0.0-r1 \
+    ca-certificates=20241121-r1 \
+    ttf-freefont=20120503-r4 \
+    xvfb=21.1.16-r0 \
+    dbus=1.14.10-r4 \
+    libx11=1.8.10-r0 \
+    libxcomposite=0.4.6-r5 \
+    libxdamage=1.1.6-r5 \
+    libxext=1.3.6-r2 \
+    && npm install -g pnpm@10.11.0
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply MemoryLeak.Dockerfile]: it works fine now without those dependencies

File: MemoryLeak.Dockerfile [hunk 37]
```diff
@@ -1,23 +1,43 @@
...
```
[the remark this answers is not in the document]
Comment (Vladomira) [reply MemoryLeak.Dockerfile]: should I keep xvfb=21.1.16-r0?

**Review by Kravalg — APPROVED** [review]
